plasma_framework/contracts/src/vaults/Vault.sol

```diff
@@ -29,4 +33,36 @@ contract Vault is Operated {
 
         framework.submitDepositBlock(root);
     }
+
+    /**
+     * @notice Sets the deposit verifier contract. This can be only called by the operator.
+     * @notice When one contract is already set next will be effective after MIN_EXIT_PERIOD.
+     * @param _contract address of the verifier contract.
+     */
+    function setDepositVerifier(address _contract) public onlyOperator {
+        require(_contract != address(0), "Cannot set an empty address as deposit verifier");
+
+        if (_currentDepositVerifier != address(0)) {
+            _newDepositVerifier = _contract;
+            _newDepositVerifierEffectivePeriod = block.timestamp + framework.minExitPeriod();
+        } else {
+            _currentDepositVerifier = _contract;
+        }
+    }
+
+    /**
+     * @notice Gets currently effective deposit verifier contract address.
+     * @return contract address of deposit verifier.
+     */
+    function getDepositVerifier() public view returns (address) {
+        require(_currentDepositVerifier != address(0), "Deposit verifier was not set yet.");
```

Comment:
I think that the tx will fail nevertheless and such a situation can only occur when the contract is being set up.
Please add a test that ensures that the deposit fails when there is no verifier, but not necessarily explicitly failed by the code.

And if we omit the check it is nothing less than a simple getter without any logic inside.

Reply:
This check only prevents the malicious operator from setting the address to zero - obviously invalid one. But it does not prevent him from setting it to any other obviously malicious ex. `address(1)`